hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SnapshotChainInfo.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership.  The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.hadoop.ozone.om;
+
+/**
+ * SnapshotChain supporting SnapshotInfo class.
+ *
+ * Defines and provides the interface to the SnapshotChainInfo
+ * entry that comprises a SnapshotChain.
+ * Getters / setters for current, prev, next element in snapshot chain.
+ */
+public class SnapshotChainInfo {
+  private String snapshotID;
```

Comment:
Should the snapshotInfo entry be stored in the SnapshotChainInfo structure instead of just the id?

Reply:
The in-memory data structure for all snapshots in the cluster consists of only a chain of strings.  Each uniquely identifying the snapshot and preserving the order of snapshots.  With this it contains only necessary info to reside in memory.  In addition we are not impacted by any changes (no dependencies) to the `SnapshotInfo` Table as `SnapshotInfo` entry fields are modified and added (schema changes).